src/main/java/ru/l1ttleO/boyfriend/settings/Settings.java

```diff
@@ -0,0 +1,255 @@
+            return this.get(key).name().toLowerCase();
+        }
+
+        @Override
+        public BotLocale putFromString(final K key, final String str) throws IllegalArgumentException {
+            final BotLocale locale;
+            try {
+                locale = BotLocale.valueOf(str.toUpperCase());
+            } catch (final IllegalArgumentException e) {
+                throw new IllegalArgumentException("Locale is unavailable: " + str);
+            }
+            put(key, locale);
+            return locale;
+        }
+
+        @Override
+        public BotLocale setFormatted(final K key, final @NotNull String str) throws IllegalArgumentException {
+            final BotLocale locale = BotLocale.valueOf(str.toUpperCase());
+            put(key, locale);
+            save(key);
+            return locale;
+        }
+    }
+
+    public static class TextChannelEntry<K> extends Entry<K, TextChannel> {
+        public TextChannelEntry(final @NotNull Function<K, Settings> keyGetter, final @NotNull String name) {
+            super(keyGetter, name, "channel", null);
+        }
+
+        @Override
+        public @Nullable String getString(final K key) {
+            final TextChannel channel = this.get(key);
+            return channel == null ? null : channel.getId();
+        }
+
+        @Override
+        public TextChannel putFromString(final K key, final String str) throws IllegalArgumentException {
+            final TextChannel channel;
+            try {
+                channel = Boyfriend.getJDA().getTextChannelById(str);
+                if (channel == null)
+                    throw new IllegalArgumentException("Unable to find channel with ID " + str);
+            } catch (final NumberFormatException e) {
+                throw new IllegalArgumentException("Invalid channel ID: " + str);
+            }
+            put(key, channel);
+            return channel;
+        }
+
+        public @Nullable String formatted(final K key) {
+            final TextChannel channel = this.get(key);
+            return channel == null ? null : channel.getAsMention();
+        }
+
+        @Override
+        public TextChannel setFormatted(final K key, final @NotNull String str) throws IllegalArgumentException {
+            final TextChannel channel = this.putFromString(key, Utils.stripChannelID(str));
+            this.save(key);
+            return channel;
+        }
+    }
+
+    public static class RoleEntry<K> extends Entry<K, Role> {
+        public RoleEntry(final @NotNull Function<K, Settings> keyGetter, final @NotNull String name) {
+            super(keyGetter, name, "role", null);
+        }
+
+        @Override
+        public @Nullable String getString(final K key) {
+            final Role role = this.get(key);
+            return role == null ? null : role.getId();
+        }
+
+        @Override
+        public Role putFromString(final K key, final String str) throws IllegalArgumentException {
```

Comment:
```suggestion
        public Role putFromString(final K key, final @NotNull String str) throws IllegalArgumentException {
```
`Overriden method parameters are not annotated` [intellij] [4/4]